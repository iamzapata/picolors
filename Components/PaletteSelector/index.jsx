import React from "react"

const PaletteSelector = ({ handlePaletteSelectChange, colorPalettes }) => (
  <div className="d-flex flex-column">
    <h6 className="text-center mb-2">Color Presets</h6>
    <div className="d-flex mb-3">
      <select className="form-select" onChange={handlePaletteSelectChange} data-testid="palette-selector">
        {colorPalettes.map((palette) => (
          <option key={palette.type} value={palette.type}>
            {palette.type}
          </option>
        ))}
      </select>
    </div>
  </div>
)

export default PaletteSelector
