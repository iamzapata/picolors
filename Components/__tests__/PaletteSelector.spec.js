import { render, screen } from '@testing-library/react'
import userEvent from '@testing-library/user-event'
import PaletteSelector from 'Components/PaletteSelector'

describe('PaletteSelector', () => {
  it('renders', () => {
    const handlePaletteSelectChange = jest.fn()
    const colorPalettes = [{ type: 'regular' }, { type: 'pastel' }]
    const { container } = render(
      <PaletteSelector
        handlePaletteSelectChange={handlePaletteSelectChange}
        colorPalettes={colorPalettes}
      />,
    )
    expect(container).toMatchSnapshot()

    userEvent.selectOptions(screen.getByTestId('palette-selector'), 'regular')

    expect(handlePaletteSelectChange).toHaveBeenCalledTimes(1)
    expect(container.querySelectorAll('option')[0].selected).toBe(true)

    userEvent.selectOptions(screen.getByTestId('palette-selector'), 'pastel')

    expect(handlePaletteSelectChange).toHaveBeenCalledTimes(2)
    expect(container.querySelectorAll('option')[1].selected).toBe(true)
  })
})
