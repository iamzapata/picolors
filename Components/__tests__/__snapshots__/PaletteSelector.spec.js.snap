// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteSelector renders 1`] = `
<div>
  <div
    class="d-flex flex-column"
  >
    <h6
      class="text-center mb-2"
    >
      Color Presets
    </h6>
    <div
      class="d-flex mb-3"
    >
      <select
        class="form-select"
        data-testid="palette-selector"
      >
        <option
          value="regular"
        >
          regular
        </option>
        <option
          value="pastel"
        >
          pastel
        </option>
      </select>
    </div>
  </div>
</div>
`;
